pandas/core/indexes/multi.py

```diff
@@ -3304,10 +3304,10 @@ def get_locs(self, seq):
             )
 
         n = len(self)
-        # indexer is the list of all positions that we want to take; we
-        #  start with it being everything and narrow it down as we look at each
-        #  entry in `seq`
-        indexer = Index(np.arange(n))
+        # indexer is the list of all positions that we want to take; it
+        #  is created on the first entry in seq and narrowed down as we
+        #  look at remaining entries
+        indexer = None
```

Comment:
same question: why does this make a difference?  IIRC this choice was made to have a consistent type for `indexer`

Reply:
the savings here is to avoid the large allocation of np.arange(n) + the intersection of that with the indexer created on the first loop. Rather than allocating and intersecting on the first loop, we just create the indexer from the first loop.

Reply:
thanks